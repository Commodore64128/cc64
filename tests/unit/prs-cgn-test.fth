
' noop alias \log

\log include logtofile.fth
\log logopen prs-cgn-test.log

  include base-setup.fth
  include scan-setup.fth

  include parser-setup.fth

  : init-fake-statics  $a000 >staticadr ;
  init: init-fake-statics

  .( GOLDEN-SECTION-START) cr

  src-begin test-global-definitions
    src@ int i = 0; @
    src@ static char x; @
    src@ extern char a /= 0x0a; @
    src@ static int e /= 0x0e; @
    src@ char *p; @
  test-begin
    T{ definition? -> true }T
    T{ definition? -> true }T
    T{ definition? -> true }T
    T{ definition? -> true }T
    T{ definition? -> true }T
    T{ definition? -> false }T
    cr

    T{ fetchglobal" i" -> $9ffe %int %extern %l-value + + }T
    T{ fetchglobal" x" -> $9ffd %l-value }T
    T{ fetchglobal" a" -> $0a %extern %l-value + }T
    T{ fetchglobal" e" -> $0e %int %l-value + }T
    T{ fetchglobal" p" -> $9ffb %extern %l-value + %pointer + }T
  test-end

  src-begin test-for
    src@ for(;;); @
    src@ {for(;;);} @
  test-begin
    T{ statement? -> true }T
    T{ statement? -> true }T
  test-end

  src-begin test-kr-func
    src@ int f(i,p) @
    src@ int i; @
    src@ char *p; @
    src@ {} @
  test-begin
    T{ definition? -> true }T
  test-end

  .( GOLDEN-SECTION-END) cr
  cr hex .( here, s0: ) here u. s0 @ u.

  cr .( test completed with ) #errors @ . .( errors) cr

\log logclose
