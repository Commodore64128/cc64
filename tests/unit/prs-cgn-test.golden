TEST-GLOBAL-DEFINITIONS
0 stat, 0 cstat, 0 stat, 
TEST-FOR
*label*: 1 
jmp ahead 2 
*label*: 3 
jmp 1 
patch 2 
jmp 3 
*label*: 4 
jmp ahead 5 
*label*: 6 
jmp 4 
patch 5 
jmp 6 
TEST-KR-FUNC
*label*: 1 
rts
TEST-INT-ARRAY-INIT
0 stat, 0 stat, 0 stat, 
3 stat, 2 stat, 1 stat, 
0 stat, 5 stat, 4 stat, 
9 stat, 8 stat, 7 stat, 
bad initializer
 static int f[2] = {10, 11, 12}; 
                               ^
11 stat, 10 stat, 
TEST-CHAR-ARRAY-INIT
0 cstat, 0 cstat, 0 cstat, 0 cstat, 
0 cstat, 67 cstat, 66 cstat, 65 cstat, 
0 cstat, 0 cstat, 69 cstat, 68 cstat, 
0 cstat, 73 cstat, 72 cstat, 71 cstat, 
bad initializer
 static char f[3] = "jkl"; 
                         ^
76 cstat, 75 cstat, 74 cstat, 
TEST-LOCAL-INIT
1 stat, 
50 cstat, 
1234 stat, 
size: 2 
lda.s #3 
size: 2 
sta.s (base),#4 

#GOLDEN TODO: Change back to size: 1 when issue/28
#GOLDEN (8/16-bit smart expressions) is solved.
size: 2 
lda.s #52 
size: 1 
sta.s (base),#6 

size: 2 
lda.s #5678 
size: 2 
sta.s (base),#7 

TEST-LOCAL-NO-INIT
0 stat, 
0 cstat, 
0 stat, 




